cmd/thanos/compact.go

```diff
@@ -121,6 +122,13 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
 	compactionConcurrency := cmd.Flag("compact.concurrency", "Number of goroutines to use when compacting groups.").
 		Default("1").Int()
 
+	deleteDelay := modelDuration(cmd.Flag("delete-delay", "Time before a block marked for deletion is deleted from bucket. "+
+		"If delete-delay is non zero, blocks will be marked for deletion and compactor component will delete blocks marked for deletion from the bucket. "+
+		"If delete-delay is 0, blocks will be deleted straight away. Use this if you want to get rid of or move the block immediately. "+
```

Comment:
This applies only for verifier and backupAndDelete (`move the block`). I would just remove `Use this if you want to get rid of or move the block immediately.`

Reply:
Copy paste error, yes the last line is not applicable here. Removed.